schema/pgsql/upgrades/032.sql

```diff
@@ -0,0 +1,100 @@
+
+CREATE INDEX idx_contactgroup_changed_at ON contactgroup(changed_at);
+
+ALTER TABLE contactgroup_member
+    ADD COLUMN changed_at bigint NOT NULL DEFAULT EXTRACT(EPOCH FROM NOW()) * 1000,
+    ADD COLUMN deleted boolenum NOT NULL DEFAULT 'n';
+
+CREATE INDEX idx_contactgroup_member_changed_at ON contactgroup_member(changed_at);
+
+ALTER TABLE schedule
+    ADD COLUMN changed_at bigint NOT NULL DEFAULT EXTRACT(EPOCH FROM NOW()) * 1000,
+    ADD COLUMN deleted boolenum NOT NULL DEFAULT 'n';
+
+CREATE INDEX idx_schedule_changed_at ON schedule(changed_at);
+
+ALTER TABLE rotation
+    ALTER COLUMN priority DROP NOT NULL,
+    ALTER COLUMN first_handoff DROP NOT NULL,
+    ADD COLUMN changed_at bigint NOT NULL DEFAULT EXTRACT(EPOCH FROM NOW()) * 1000,
+    ADD COLUMN deleted boolenum NOT NULL DEFAULT 'n',
+    ADD CHECK (deleted = 'y' OR priority IS NOT NULL AND first_handoff IS NOT NULL);
+
+CREATE INDEX idx_rotation_changed_at ON rotation(changed_at);
+
+ALTER TABLE timeperiod
+    ADD COLUMN changed_at bigint NOT NULL DEFAULT EXTRACT(EPOCH FROM NOW()) * 1000,
+    ADD COLUMN deleted boolenum NOT NULL DEFAULT 'n';
+
+CREATE INDEX idx_timeperiod_changed_at ON timeperiod(changed_at);
+
+ALTER TABLE rotation_member
+    RENAME CONSTRAINT rotation_member_check TO ck_rotation_member_either_contact_id_or_contactgroup_id;
+
+ALTER TABLE rotation_member
+    ALTER COLUMN position DROP NOT NULL,
+    ADD COLUMN changed_at bigint NOT NULL DEFAULT EXTRACT(EPOCH FROM NOW()) * 1000,
+    ADD COLUMN deleted boolenum NOT NULL DEFAULT 'n',
+    ADD CONSTRAINT ck_rotation_member_non_deleted_needs_position CHECK (deleted = 'y' OR position IS NOT NULL);
+
+CREATE INDEX idx_rotation_member_changed_at ON rotation_member(changed_at);
+
+ALTER TABLE timeperiod_entry
+    ADD COLUMN changed_at bigint NOT NULL DEFAULT EXTRACT(EPOCH FROM NOW()) * 1000,
+    ADD COLUMN deleted boolenum NOT NULL DEFAULT 'n';
+
+CREATE INDEX idx_timeperiod_entry_changed_at ON timeperiod_entry(changed_at);
+
+ALTER TABLE source
+    ADD COLUMN changed_at bigint NOT NULL DEFAULT EXTRACT(EPOCH FROM NOW()) * 1000,
+    ADD COLUMN deleted boolenum NOT NULL DEFAULT 'n';
+
+CREATE INDEX idx_source_changed_at ON source(changed_at);
+
+ALTER TABLE rule
+    ADD COLUMN changed_at bigint NOT NULL DEFAULT EXTRACT(EPOCH FROM NOW()) * 1000,
+    ADD COLUMN deleted boolenum NOT NULL DEFAULT 'n';
+
+CREATE INDEX idx_rule_changed_at ON rule(changed_at);
+
+UPDATE rule SET deleted = 'y' WHERE is_active = 'n';
+ALTER TABLE rule DROP COLUMN is_active;
+
+ALTER TABLE rule_escalation
+    RENAME CONSTRAINT rule_escalation_check TO ck_rule_escalation_not_both_condition_and_fallback_for;
+
+ALTER TABLE rule_escalation
+    ALTER COLUMN position DROP NOT NULL,
+    ADD COLUMN changed_at bigint NOT NULL DEFAULT EXTRACT(EPOCH FROM NOW()) * 1000,
+    ADD COLUMN deleted boolenum NOT NULL DEFAULT 'n',
+    ADD CONSTRAINT ck_rotation_member_non_deleted_needs_position CHECK (deleted = 'y' OR position IS NOT NULL),
```

Comment:
Still not quite the right constraint name :see_no_evil:

Diff between freshly imported and upgraded schema:
```diff
 CREATE TABLE public.rule_escalation (
     id bigint NOT NULL,
     rule_id bigint NOT NULL,
     "position" integer,
     condition text,
     name public.citext,
     fallback_for bigint,
     changed_at bigint DEFAULT (EXTRACT(epoch FROM now()) * (1000)::numeric) NOT NULL,
     deleted public.boolenum DEFAULT 'n'::public.boolenum NOT NULL,
-    CONSTRAINT ck_rule_escalation_non_deleted_needs_position CHECK (((deleted = 'y'::public.boolenum) OR ("position" IS NOT NULL))),
+    CONSTRAINT ck_rotation_member_non_deleted_needs_position CHECK (((deleted = 'y'::public.boolenum) OR ("position" IS NOT NULL))),
     CONSTRAINT ck_rule_escalation_not_both_condition_and_fallback_for CHECK ((NOT ((condition IS NOT NULL) AND (fallback_for IS NOT NULL))))
 );
```

Reply:
🫠